src/ipDeviceCommon/MemoryPool.h

```diff
@@ -0,0 +1,99 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#ifndef _RS_MEMORY_POOL_HH
+#define _RS_MEMORY_POOL_HH
+
+#include <queue>
+#include <mutex>
+#include <iostream>
+#include <ipDeviceCommon/RsCommon.h>
+
+#define POOL_SIZE 100                 //TODO:: to define the right value
+#define MAX_FRAME_SIZE 1280 * 720 * 3 //TODO:: to define the right value
```

Comment:
I would recommend against using this memory pool implementation - max frame size can be larger in near future, and it's significant amount of memory to hold just in case. 
Also, librealsense already has memory pool.
I recommend to use new / delete for now and eventually port to librealsense mem-pool in the future
not blocker

Reply:
W'll handle it after the first version